### DISCUSSION TITLE
Finding defined aliases from a package's commands section

### DISCUSSION THREAD

**Discussion by herronelou** [AcademySoftwareFoundation/rez#1400] [Q&A] [answered]
Hello.

I'm finding myself in a situation where I need to define an alias which will in turn call another alias.
Now, this is not possible with doskey in windows cmd, but I was wondering if maybe it is possible to find already-defined aliases during the `commands` section so that I could "expand" my new alias at creation.

### A more concrete example:
My Nuke package defines an alias `nuke`, which points at the right nuke exe (for example `Nuke13.1.exe` for Nuke13.1 on windows).
My other package `foo` needs to execute a nuke command line operation: `nuke -t do-stuff.py my_file.exr`, this command is too long so I would like to set `alias('nuke -t do-stuff.py', 'do_stuff')`.
Foo, doesn't necessarily know which Nuke was resolved, so it doesn't know how to call `Nuke13.1.exe -t do_struff.py`, but using `nuke` directly doesn't work either because doskey will not resolve nested aliases.
Is there some real code that would do something similar to this pseudo_code:
```
nuke_alias = self.aliases.get('nuke')
alias('do_stuff', '{} -t do_struff.py'.format(nuke_alias)
```

I feel like I could get around that possibly by creating a .bat file as an intermediary step (one alias pointing at the .bat, and the .bat using the other alias), and I'm about to try if this works, but the downside of that is that .bat files are windows only, and I'd have to make variants for other OSes if going this route.  Edit: Can't use doskey aliases in .bat files apparently, so that's not the way either.

**Comment by instinct-vfx** [comment, marked as answer]
Note that this can become really really tricky, specifically in cmd and specifically if you need to do things like using environment variables as parameters in a bat file. You can end up in escape madness quickly.

That said what about setting an env variable to the value of the alias when setting the alias and then looking for the env variable?

**Comment by herronelou** [comment]
Yeah I landed on setting a NUKE_EXECUTABLE env var on the nuke package. Not
loving setting specific stuff upstream for downstream packages but I
couldn't quite think of a solution that wouldn't be less interdependent.
